src/utransport.rs

```diff
@@ -85,5 +83,327 @@ pub trait UTransport {
     /// # Errors
     ///
     /// Returns an error if the listener could not be unregistered, for example if the given listener does not exist.
-    async fn unregister_listener(&self, topic: UUri, listener: &str) -> Result<(), UStatus>;
+    async fn unregister_listener<T>(&self, topic: UUri, listener: Arc<T>) -> Result<(), UStatus>
+    where
+        T: UListener;
+}
+
+/// A wrapper type around UListener that can be used by `up-client-foo-rust` UPClient libraries
+/// to ease some common development scenarios when we want to compare `UListener`
+///
+/// # Note
+///
+/// Not necessary for end-user uEs to use. Primarily intended for `up-client-foo-rust` UPClient libraries
+///
+/// # Rationale
+///
+/// The wrapper type is implemented such that it can be used in any location you may wish to
+/// hold a generic `UListener`
+///
+/// Also implements necessary traits to allow hashing, so that you may hold the wrapper type in
+/// collections which require that, such as a `HashMap` or `HashSet`
+///
+/// Note that the implementation is such that if the same instance of `Arc<T>` is used multiple times
```

Comment:
actually, it is not about the same *instance* of Arc but about the UListener implementation they point to, i.e. it is about the Arc's *pointer*.